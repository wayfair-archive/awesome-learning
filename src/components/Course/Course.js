import React from "react";
import PropTypes from "prop-types";
import Tags from "./Tags";
import StyledLink from "../Link";
import CourseCard from "../CourseCard";
import ContentSection from "../ContentSection";
import "./Course.scss";

const Course = ({ course }) => {
  const { tags, title, description, lessons } = course.frontmatter;
  const { tagSlugs, slug } = course.fields;
  return (
    <div className="Course">
      <div className="Course-backLink">
        <StyledLink variation={"tertiary"} path={"/courses"}>
          Back to Courses
        </StyledLink>
      </div>
      <ContentSection title={title}>
        <p>{description}</p>
      </ContentSection>
      <ul>
        {lessons.map(({ link, title, description }) => {
          const composedLink = `${slug}${link}`;
          return (
            <CourseCard
              key={title}
              title={title}
              link={composedLink}
              description={description}
            />
          );
        })}
      </ul>
      {tags.length > 2 && (
        <React.Fragment>
          <div className="Course-footer" />
          <ContentSection title="Related Themes">
            <Tags tags={tags} tagSlugs={tagSlugs} />
          </ContentSection>
        </React.Fragment>
      )}
    </div>
  );
};

Course.propTypes = {
  course: PropTypes.shape({
    frontmatter: PropTypes.shape({
      tags: PropTypes.array,
      title: PropTypes.string,
      description: PropTypes.node,
      lessons: PropTypes.arrayOf(PropTypes.object)
    }).isRequired,
    fields: PropTypes.shape({
      tagSlugs: PropTypes.array,
      slug: PropTypes.string
    })
  })
}

export default Course;
