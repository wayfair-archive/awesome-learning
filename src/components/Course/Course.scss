@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.Course {
  &-backLink {
    @include margin(1, 1.3, 0, 1.3);
  }
  &-footer {
    border-top: 1px solid $color--border;
    @include margin(2, 1.3 ,0 ,1.3 )
  }
}