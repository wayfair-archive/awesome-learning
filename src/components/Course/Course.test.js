import React from 'react';
import renderer from 'react-test-renderer';
import Course from './Course';

describe('Course', () => {
  const props = {
    course: {
      html: '<p>test</p>',
      slug: '/slug',
      fields: {
        tagSlugs: [
          '/test_0',
          '/test_1'
        ]
      },
      frontmatter: {
        date: '2016-09-01',
        tags: [
          'test_0',
          'test_1'
        ],
        title: 'test',
        lessons: [{
          link: 'link',
          title: 'this is the title',
          description: 'this is the description'
        }]
      }
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Course {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
