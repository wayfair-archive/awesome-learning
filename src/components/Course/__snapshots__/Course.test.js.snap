// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course renders correctly 1`] = `
<div
  className="Course"
>
  <div
    className="Course-backLink"
  >
    <Link
      activeClassName=""
      className="Link Link--tertiary"
      to="/courses"
    >
      Back to Courses
    </Link>
  </div>
  <div
    className="ContentSection"
  >
    <header
      className="ContentSection-title"
    >
      <h1>
        test
      </h1>
    </header>
    <div
      className="ContentSection-content  ContentSection-content--alignLeft"
    >
      <p />
    </div>
  </div>
  <ul>
    <li
      className="CourseCard"
    >
      <h3
        data-enzyme-id="Block-RenderedElement"
        style={
          Object {
            "marginBottom": "16px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
          }
        }
      >
        this is the title
      </h3>
      <p
        data-enzyme-id="Block-RenderedElement"
        style={
          Object {
            "marginBottom": "16px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
          }
        }
      >
        this is the description
      </p>
      <Link
        activeClassName=""
        className="Link Link--pill"
        to="undefinedlink"
      >
        Learn 
        this is the title
      </Link>
    </li>
  </ul>
</div>
`;
